config/appstructure.go

```diff
@@ -40,10 +49,36 @@ type (
 		report        AppStructureReport
 		structureLock sync.Mutex
 	}
+
+	// Appstructure and warnings in consumable form
+	Appstructure struct {
+		Guid      string
+		Warnings  []AppStructureWarning
+		Structure *appstructure.AppStructure
+	}
 )
 
-func (cfg *Config) getAppStructureScenario(includeRaw bool, summary SummaryType) []scenario.Action {
-	return evaluateActionList(cfg.Scenario, includeRaw, summary)
+const (
+	AppStructureWarningObjectNotExists          = "ObjectNotExists"
+	AppStructureWarningMasterObjectNotExists    = "MasterObjectNotExists"
+	AppStructureWarningUnsupportedVisualization = "UnsupportedVisualization"
+	AppStructureWarningMissingDimMeasure        = "MissingDimMeasure"
+	AppStructureWarningMeasureNotFound          = "MeasureNotFound"
+	AppStructureWarningMetaInformationNotFound  = "MetaInformationNotFound"
+	AppStructureWarningDimensionNotFound        = "DimensionNotFound"
+	AppStructureWarningPropertiesError          = "PropertiesError"
+	AppStructureWarningChildError               = "ChildError"
+	AppStructureWarningSnapshotError            = "SnapshotError"
+)
+
+// String implements Stringer interface
+func (warning AppStructureWarning) String() string {
+	return fmt.Sprintf("object<%s> type<%s> visualization<%s> warning: %s", warning.ObjectID, warning.ObjectType, warning.ObjectVisualization, warning.Message)
+}
+
+func (cfg *Config) getAppStructureScenario(includeRaw bool) ([]scenario.Action, map[string]*GeneratedAppStructure) {
+	appStructureMap := make(map[string]*GeneratedAppStructure, 1)
```

Comment:
Why is this initialized with explicit capacity 1?

Reply:
just because it's the most common size which will be used. (unless running script structure command with a scenario containing multiple openapp actions). You think it'd be better to keep lazy grow?

Reply:
Alright, sounds reasonable 👍